lua/mediaplayer/services/youtube/init.lua

```diff
@@ -134,18 +135,96 @@ function SERVICE:GetMetadata( callback )
 		callback(self._metadata)
 
 	else
-
 		local videoId = self:GetYouTubeVideoId()
-		local apiurl = MetadataUrl:format( videoId, APIKey )
+		local videoUrl = "https://www.youtube.com/watch?v="..videoId
 
-		self:Fetch( apiurl,
+		self:Fetch( videoUrl,
+			--On Success
 			function( body, length, headers, code )
-				OnReceiveMetadata( self, callback, body )
+				local metadata = self:ParseYTMetaDataFromHTML(body, videoId)
+
+				--html couldn't be parsed
+				if (!metadata.title || !metadata.duration) then
```

Comment:
I think this will fail for live streams where `duration == 0`.

It should just need a type check.
```suggestion
				if (!metadata.title || !isnumber(metadata.duration)) then
```